e2e/tests/wasm/grandpa_test.go

```diff
@@ -0,0 +1,407 @@
+	rep := testreporter.NewReporter(f)
+	eRep := rep.RelayerExecReporter(t)
+
+	ctx := context.Background()
+
+
+
+	nv := 2 // Number of validators
+	nf := 1 // Number of full nodes
+
+	if testsuite.IsCI() {
+		nv = 5
+	}
+
+	consensusOverrides := make(testutil.Toml)
+	blockTime := 5 // seconds, parachain is 12 second blocks, don't make relayer work harder than needed
+	blockT := (time.Duration(blockTime) * time.Second).String()
+	consensusOverrides["timeout_commit"] = blockT
+	consensusOverrides["timeout_propose"] = blockT
+
+	configTomlOverrides := make(testutil.Toml)
+	configTomlOverrides["consensus"] = consensusOverrides
+
+	configFileOverrides := make(map[string]any)
+	configFileOverrides["config/config.toml"] = configTomlOverrides
+
+	// Get both chains
+	cf := interchaintest.NewBuiltinChainFactory(zaptest.NewLogger(t), []*interchaintest.ChainSpec{
+		{
+			ChainName: "composable", // Set ChainName so that a suffix with a "dash" is not appended (required for hyperspace)
+			ChainConfig: ibc.ChainConfig{
+				Type:    "polkadot",
+				Name:    "composable",
+				ChainID: "rococo-local",
+				Images: []ibc.DockerImage{
+					{
+						Repository: "ghcr.io/misko9/polkadot-node",
+						Version:    "local",
+						UidGid:     "1000:1000",
+					},
+					{
+						Repository: "ghcr.io/misko9/parachain-node",
+						Version:    "latest",
+						UidGid:     "1000:1000",
+					},
+				},
+				Bin:            "polkadot",
+				Bech32Prefix:   "composable",
+				Denom:          "uDOT",
+				GasPrices:      "",
+				GasAdjustment:  0,
+				TrustingPeriod: "",
+				CoinType:       "354",
+			},
+			NumValidators: &nv,
+			NumFullNodes:  &nf,
+		},
+		{
+			ChainName: "simd", // Set chain name so that a suffix with a "dash" is not appended (required for hyperspace)
+			ChainConfig: ibc.ChainConfig{
+				Type:    "cosmos",
+				Name:    "simd",
+				ChainID: "simd",
+				Images: []ibc.DockerImage{
+					{
+						Repository: "ghcr.io/misko9/ibc-go-simd",
+						Version:    "local",
+						UidGid:     "1025:1025",
+					},
+				},
+				Bin:            "simd",
+				Bech32Prefix:   "cosmos",
+				Denom:          "stake",
+				GasPrices:      "0.00stake",
+				GasAdjustment:  1.3,
+				TrustingPeriod: "504h",
+				CoinType:       "118",
+				NoHostMount:         true,
+				ConfigFileOverrides: configFileOverrides,
+				ModifyGenesis:       modifyGenesisShortProposals(votingPeriod, maxDepositPeriod),
+			},
+		},
+	})
+
+	chains, err := cf.Chains(t.Name())
+	s.Require().NoError(err)
+
+	polkadotChain := chains[0].(*polkadot.PolkadotChain)
+	cosmosChain := chains[1].(*cosmos.CosmosChain)
+
+	// Get a relayer instance
+	r := interchaintest.NewBuiltinRelayerFactory(
+		ibc.Hyperspace,
+		zaptest.NewLogger(t),
+		relayer.ImagePull(true),
+		relayer.CustomDockerImage("ghcr.io/misko9/hyperspace", "local", "1000:1000"),
+	).Build(t, client, network)
+
+	// Build the network; spin up the chains and configure the relayer
+	const pathName = "composable-simd"
```

Comment:
move at const block?

Reply:
I'll pull this out in the followup if that's okay with you!